backends/ebpf/ebpftargets.py

```diff
@@ -0,0 +1,343 @@
+#!/usr/bin/env python
+# Copyright 2018 VMware, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Contains different eBPF models and specifies their individual behavior
+# Currently five phases are defined:
+#   1. Invokes the specified compiler on a provided p4 file.
+#   2. Parses an stf file and generates an pcap output.
+#   3. Loads the generated template or compiles it to a runnable binary.
+#   4. Feeds the generated pcap test packets into the P4 "switch"
+#   5. Evaluates the output with the expected result from the .stf file
+
+from __future__ import print_function
+from subprocess import Popen
+from threading import Thread
+from glob import glob
+try:
+    from scapy.layers.all import *
+    from scapy.utils import *
+except ImportError:
+    pass
+
+SUCCESS = 0
+FAILURE = 1
+TIMEOUT = 10 * 60
+
+
+def nextWord(text, sep=None):
+    # Split a text at the indicated separator.
+    # Note that the separator can be a string.
+    # Separator is discarded.
+    spl = text.split(sep, 1)
+    if len(spl) == 0:
+        return '', ''
+    elif len(spl) == 1:
+        return spl[0].strip(), ''
+    else:
+        return spl[0].strip(), spl[1].strip()
+
+
+def ByteToHex(byteStr):
+    return ''.join(["%02X " % ord(x) for x in byteStr]).strip()
+
+
+def HexToByte(hexStr):
+    bytes = []
+    hexStr = ''.join(hexStr.split(" "))
+    for i in range(0, len(hexStr), 2):
+        bytes.append(chr(int(hexStr[i:i + 2], 16)))
+    return ''.join(bytes)
+
+
+def isError(p4filename):
```

Comment:
There are two tests that fail right now hit and two. Largely because the compiler seems to be generating wrong output. I added these to the xfail tests for now.

Reply:
This has nothing to do with xfail; these are programs that are incorrect syntactically.
The tests will succeed, because we will invert the compiler output result. They are PASS, not XFAIL.